{"repository":"https://github.com/tommybananas/finale","dockerfilePath":"Dockerfile","startTime":1672833411709,"clone":{"stdout":"","stderr":"Cloning into '/tmp/dinghy-analysis/tommybananas/finale'...\n","error":null,"commit":"d94ee1f4cc03289f78737ff25b71522ccafbafcb"},"originalBuild":{"startTimestamp":1672833412650,"endTimestamp":1672833443067,"error":null,"stdout":"Sending build context to Docker daemon    638kB\r\r\nStep 1/7 : FROM node:10.24.1-alpine\n10.24.1-alpine: Pulling from library/node\nddad3d7c1e96: Already exists\nde915e575d22: Pulling fs layer\n7150aa69525b: Pulling fs layer\nd7aa47be044e: Pulling fs layer\nd7aa47be044e: Download complete\n7150aa69525b: Verifying Checksum\n7150aa69525b: Download complete\nde915e575d22: Verifying Checksum\nde915e575d22: Download complete\nde915e575d22: Pull complete\n7150aa69525b: Pull complete\nd7aa47be044e: Pull complete\nDigest: sha256:dc98dac24efd4254f75976c40bce46944697a110d06ce7fa47e7268470cf2e28\nStatus: Downloaded newer image for node:10.24.1-alpine\n ---> aa67ba258e18\nStep 2/7 : RUN mkdir -p /usr/src/app\n ---> Running in df5897c701a8\nRemoving intermediate container df5897c701a8\n ---> 4a52201cdaf3\nStep 3/7 : WORKDIR /usr/src/app\n ---> Running in 8a2166109d4a\nRemoving intermediate container 8a2166109d4a\n ---> 8974a68bc045\nStep 4/7 : COPY package*.json /usr/src/app/\n ---> 8a8f8c41026e\nStep 5/7 : RUN npm install && npm cache clean --force\n ---> Running in 3aebd3bf232e\n\n> dtrace-provider@0.8.8 install /usr/src/app/node_modules/dtrace-provider\n> node-gyp rebuild || node suppress-error.js\n\n\u001b[91mgyp\u001b[0m\u001b[91m ERR! find Python \ngyp ERR!\u001b[0m\u001b[91m find Python Python is not set from command line or npm configuration\ngyp ERR! find Python Python is not set from environment variable PYTHON\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python checking if \"python\" can be used\n\u001b[0m\u001b[91mgyp ERR! find Python - \"python\" is not in PATH or produced an error\ngyp ERR!\u001b[0m\u001b[91m find Python checking if \"python2\" can be used\ngyp ERR!\u001b[0m\u001b[91m find Python - \"python2\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python checking if \"python3\" can be used\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m - \"python3\" is not in PATH or produced an error\ngyp\u001b[0m\u001b[91m ERR! find Python \ngyp\u001b[0m\u001b[91m ERR! find Python\u001b[0m\u001b[91m **********************************************************\ngyp \u001b[0m\u001b[91mERR! find Python You need to install the latest version of Python.\ngyp ERR!\u001b[0m\u001b[91m find Python Node-gyp should be able to find and use Python. If not,\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! find Python\u001b[0m\u001b[91m you can try one of the following options:\ngyp \u001b[0m\u001b[91mERR! find Python - Use the switch --python=\"/path/to/pythonexecutable\"\n\u001b[0m\u001b[91mgyp ERR! find Python   (accepted by both node-gyp and npm)\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python - Set the environment variable PYTHON\ngyp \u001b[0m\u001b[91mERR! find Python\u001b[0m\u001b[91m - Set the npm configuration variable python:\ngyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR! find Python\u001b[0m\u001b[91m   npm config set python \"/path/to/pythonexecutable\"\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mfind Python For more information consult the documentation at:\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m find Python https://github.com/nodejs/node-gyp#installation\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python **********************************************************\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python \n\u001b[0m\u001b[91mgyp ERR! configure error \n\u001b[0m\u001b[91mgyp ERR! stack Error: Could not find any Python installation to use\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)\ngyp \u001b[0m\u001b[91mERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m stack\u001b[0m\u001b[91m     at exithandler (child_process.js:301:5)\ngyp \u001b[0m\u001b[91mERR! stack     at ChildProcess.errorhandler (child_process.js:313:5)\ngyp ERR!\u001b[0m\u001b[91m stack     at ChildProcess.emit (events.js:198:13)\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mstack     at Process.ChildProcess._handle.onexit (internal/child_process.js:246:12)\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR! stack     at onErrorNT (internal/child_process.js:415:16)\ngyp ERR! stack     at process._tickCallback (internal/process/next_tick.js:63:19)\n\u001b[0m\u001b[91mgyp ERR! System\u001b[0m\u001b[91m Linux 5.4.0-113-generic\n\u001b[0m\u001b[91mgyp ERR! command\u001b[0m\u001b[91m \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mcwd\u001b[0m\u001b[91m /usr/src/app/node_modules/dtrace-provider\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m node -v v10.24.1\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mnode-gyp -v v5.1.0\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mnot ok\u001b[0m\u001b[91m \n\u001b[0m\n> sqlite3@4.2.0 install /usr/src/app/node_modules/sqlite3\n> node-pre-gyp install --fallback-to-build\n\n\u001b[91mnode-pre-gyp\u001b[0m\u001b[91m WARN \u001b[0m\u001b[91mUsing request for node-pre-gyp https download \n\u001b[0m[sqlite3] Success: \"/usr/src/app/node_modules/sqlite3/lib/binding/node-v64-linux-x64/node_sqlite3.node\" is installed via remote\nadded 313 packages from 858 contributors and audited 313 packages in 7.441s\n\n13 packages are looking for funding\n  run `npm fund` for details\n\nfound 16 vulnerabilities (5 moderate, 5 high, 6 critical)\n  run `npm audit fix` to fix them, or `npm audit` for details\n\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mWARN\u001b[0m\u001b[91m using --force\u001b[0m\u001b[91m I sure hope you know what you are doing.\n\u001b[0mRemoving intermediate container 3aebd3bf232e\n ---> 466b87900a0c\nStep 6/7 : COPY . /usr/src/app\n ---> d4f8b6f7cec1\nStep 7/7 : CMD [ \"npm\", \"test\" ]\n ---> Running in f0a9570a059e\nRemoving intermediate container f0a9570a059e\n ---> b255b7e90dba\nSuccessfully built b255b7e90dba\nSuccessfully tagged finale:latest\n","stderr":"","imageSize":127373517},"repair":{"violations":[{"name":"mkdirUsrSrcThenRemove","position":{"start":2,"end":2,"columnStart":4,"columnEnd":25}}],"repairedDockerfile":"FROM node:10.24.1-alpine\n\nRUN mkdir -p /usr/src/app\nWORKDIR /usr/src/app\n\nCOPY package*.json /usr/src/app/\nRUN npm install && npm cache clean --force\n\n#ONBUILD ARG NODE_ENV\n#ONBUILD ENV NODE_ENV $NODE_ENV\nCOPY . /usr/src/app\n\nCMD [ \"npm\", \"test\" ]"},"repairedBuild":{"startTimestamp":1672833445087,"endTimestamp":1672833465824,"error":null,"stdout":"Sending build context to Docker daemon    638kB\r\r\nStep 1/7 : FROM node:10.24.1-alpine\n ---> aa67ba258e18\nStep 2/7 : RUN mkdir -p /usr/src/app\n ---> Running in 1932e54dc59d\nRemoving intermediate container 1932e54dc59d\n ---> 2d80ed1ff141\nStep 3/7 : WORKDIR /usr/src/app\n ---> Running in 8c674bb1a92e\nRemoving intermediate container 8c674bb1a92e\n ---> 9c7123700110\nStep 4/7 : COPY package*.json /usr/src/app/\n ---> b33a4eb126ce\nStep 5/7 : RUN npm install && npm cache clean --force\n ---> Running in 42aa62fb9020\n\n> dtrace-provider@0.8.8 install /usr/src/app/node_modules/dtrace-provider\n> node-gyp rebuild || node suppress-error.js\n\n\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m \ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python Python is not set from command line or npm configuration\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m Python is not set from environment variable PYTHON\ngyp\u001b[0m\u001b[91m ERR! find Python checking if \"python\" can be used\ngyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python - \"python\" is not in PATH or produced an error\ngyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m checking if \"python2\" can be used\ngyp ERR!\u001b[0m\u001b[91m find Python - \"python2\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m checking if \"python3\" can be used\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python - \"python3\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m \ngyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python **********************************************************\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m You need to install the latest version of Python.\ngyp ERR! find Python Node-gyp should be able to find and use Python. If not,\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m you can try one of the following options:\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m - Use the switch --python=\"/path/to/pythonexecutable\"\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python   (accepted by both node-gyp and npm)\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m find Python - Set the environment variable PYTHON\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m - Set the npm configuration variable python:\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m   npm config set python \"/path/to/pythonexecutable\"\ngyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m For more information consult the documentation at:\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m https://github.com/nodejs/node-gyp#installation\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python **********************************************************\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m \n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m configure error\u001b[0m\u001b[91m \n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mstack\u001b[0m\u001b[91m Error: Could not find any Python installation to use\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mstack\u001b[0m\u001b[91m     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)\ngyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR! stack\u001b[0m\u001b[91m     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m stack\u001b[0m\u001b[91m     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)\ngyp ERR! stack\u001b[0m\u001b[91m     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)\ngyp ERR! \u001b[0m\u001b[91mstack     at exithandler (child_process.js:301:5)\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mstack     at ChildProcess.errorhandler (child_process.js:313:5)\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mstack\u001b[0m\u001b[91m     at ChildProcess.emit (events.js:198:13)\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mstack\u001b[0m\u001b[91m     at Process.ChildProcess._handle.onexit (internal/child_process.js:246:12)\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mstack     at onErrorNT (internal/child_process.js:415:16)\ngyp \u001b[0m\u001b[91mERR! stack\u001b[0m\u001b[91m     at process._tickCallback (internal/process/next_tick.js:63:19)\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mSystem Linux 5.4.0-113-generic\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mcommand \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mcwd\u001b[0m\u001b[91m /usr/src/app/node_modules/dtrace-provider\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m node -v\u001b[0m\u001b[91m v10.24.1\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m node-gyp -v\u001b[0m\u001b[91m v5.1.0\ngyp\u001b[0m\u001b[91m ERR! not ok\u001b[0m\u001b[91m \n\u001b[0m\n> sqlite3@4.2.0 install /usr/src/app/node_modules/sqlite3\n> node-pre-gyp install --fallback-to-build\n\n\u001b[91mnode-pre-gyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mWARN Using request for node-pre-gyp https download \n\u001b[0m[sqlite3] Success: \"/usr/src/app/node_modules/sqlite3/lib/binding/node-v64-linux-x64/node_sqlite3.node\" is installed via remote\nadded 313 packages from 858 contributors and audited 313 packages in 5.638s\n\n13 packages are looking for funding\n  run `npm fund` for details\n\nfound 16 vulnerabilities (5 moderate, 5 high, 6 critical)\n  run `npm audit fix` to fix them, or `npm audit` for details\n\u001b[91mnpm\u001b[0m\u001b[91m WARN using --force I sure hope you know what you are doing.\n\u001b[0mRemoving intermediate container 42aa62fb9020\n ---> aebb29ef93c5\nStep 6/7 : COPY . /usr/src/app\n ---> 31966cef961d\nStep 7/7 : CMD [ \"npm\", \"test\" ]\n ---> Running in 462b1dff15eb\nRemoving intermediate container 462b1dff15eb\n ---> 5e72ae39f97c\nSuccessfully built 5e72ae39f97c\nSuccessfully tagged finale:latest\n","stderr":"","imageSize":127373517},"endTime":1672833467708}